File: typescript/src/Basic_Patterns/Behavioural_Patterns/Visitor/Geometric_Shapes/code/elements/Dot.ts
import {Visitor} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Geometric_Shapes/code/visitors/Visitor";
import {Shape} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Geometric_Shapes/code/elements/Shape";

/** Concrete Element */
class Dot implements Shape {
    move(x: number, y: number): void {
        // move shape
    }

    draw(): void {
        // draw shape
    }

    accept(visitor: Visitor): void {
        visitor.visitDot(this);
    }
}

export { Dot };

File: java/src/Basic_Patterns/Behavioural_Patterns/Double_Dispatch/Problem_Showcase/Elements/Dot.java
package Basic_Patterns.Behavioural_Patterns.Double_Dispatch.Problem_Showcase.Elements;

public class Dot extends Shape {
    protected int x, y;

    public Dot(String id, int x, int y) {
        super(id);
        this.x = x;
        this.y = y;
    }

    @Override
    public void draw() {
        System.out.println("Drawing a dot at (" + x + ", " + y + ").");
    }
}
